### PR TITLE
Add support for CSI Blob drivers

### PR DESCRIPTION
**Pull request by melzayet**
## PR Summary

Implements #342
Added an add-on section for CSI Azure blob in addonsTab.js
Added a param and profile section in main.bicep
Added deployment param setting in deployTab.js
Added a default in config.json

image

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] This PR is ready to merge and is not **Work in Progress**
- [x] Link to a filed issue
- [x] Screenshot of UI changes (if PR includes UI changes)

### CODE REVIEW

File: helper/src/components/addonsTab.js [hunk 1]
```diff
@@ -422,6 +422,22 @@ export default function ({ tabValues, updateFn, featureFlag, invalidArray }) {
                     label="Enable high availability mode"
                 />
             </Stack.Item>
+
+            <Separator className="notopmargin" />
+
+            <Stack.Item align="start">
+                <Label required={true}>
+                    Azure Blob storage CSI (Container Storage Interface - preview): Enable blob access from the cluster using BlobFuse or Network File System (NFS) version 3.0 protocol
+                    (<a target="_new" href="https://docs.microsoft.com/en-gb/azure/aks/azure-blob-csi">docs</a>)
```
Comment (Gordonby) [inline helper/src/components/addonsTab.js:431]: Can you take out "en-gb/" - we're trying to use non regional page references

File: helper/src/components/deployTab.js [hunk 2]
```diff
@@ -66,6 +66,7 @@ export default function DeployTab({ defaults, updateFn, tabValues, invalidArray,
     ...(defaults.addons.openServiceMeshAddon !== addons.openServiceMeshAddon && {openServiceMeshAddon: addons.openServiceMeshAddon }),
     ...(addons.azurepolicy !== "none" && { azurepolicy: addons.azurepolicy }),
     ...(addons.azurepolicy !== "none" && addons.azurePolicyInitiative !== defaults.addons.azurePolicyInitiative && { azurePolicyInitiative: addons.azurePolicyInitiative }),
+    ...(defaults.addons.blobCSIAddon !== addons.blobCSIAddon && {blobCSIAddon: addons.blobCSIAddon }),
```
Comment (Gordonby) [inline helper/src/components/deployTab.js:69]: As blob csi is a preview feature, can you follow these steps;
https://github.com/Azure/AKS-Construction/blob/main/docs/ContributingWizard.md#preview-features

**Review by Gordonby — CHANGES_REQUESTED** [review]

File: helper/src/components/addonsTab.js (hunk 1 above)
Comment (melzayet) [reply helper/src/components/addonsTab.js]: done!

**Review by Gordonby — APPROVED** [review]

**Comment by khowling** [comment]
Looks like a great addition @melzayet, we haven't done much around attached storage so far.  Just couple of comments to consider, we have a CSI driver for Secrets,  should we keep the 'CSI' stuff together in the UI? I'm not sure though

File: helper/src/components/addonsTab.js [hunk 3]
```diff
@@ -422,6 +422,22 @@ export default function ({ tabValues, updateFn, featureFlag, invalidArray }) {
                     label="Enable high availability mode"
                 />
             </Stack.Item>
+
```
Comment (Gordonby) [inline helper/src/components/addonsTab.js:425]: Can you bump this further up the page, between CSI Secrets and KEDA?

File: helper/src/components/addonsTab.js [hunk 4]
```diff
@@ -422,6 +422,22 @@ export default function ({ tabValues, updateFn, featureFlag, invalidArray }) {
...
+            <Separator className="notopmargin" />
+
+            <Stack.Item align="start">
+                <Label required={true}>
+                    Azure Blob storage CSI driver: Enable BlobFuse or NFS v3 access to Azure Blob
```
Comment (Gordonby) [inline helper/src/components/addonsTab.js:430]: Can you rename the label to be `CSI Blob storage: Enable BlobFuse or NFS v3 access to Azure Blob Storage`

**Review by Gordonby — CHANGES_REQUESTED** [review]

**Comment by melzayet** [comment]
@khowling indeed better to group CSI items close. I have kept them grouped without a separator but happy to separate them if looks better

Also could not find limitations or conflicts with other settings

@Gordonby 2 changes done

**Review by Gordonby — APPROVED** [review]
lgtm

**Pull request merged**